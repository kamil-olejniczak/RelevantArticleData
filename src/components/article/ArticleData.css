.ArticleData {
    background-color: #EADEDB;
    border-radius: 5px;
    padding: 10px;
    margin: 20px 0 0;
}

.ArticleData p {
     margin-top: 0;
 }
