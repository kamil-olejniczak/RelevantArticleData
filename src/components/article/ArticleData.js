import React from 'react';

const ArticleData = () => (
    <div>ArticleData is working!</div>
);

export default ArticleData;
