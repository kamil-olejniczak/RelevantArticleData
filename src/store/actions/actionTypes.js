export const INIT_GET_ARTICLE = 'INIT_GET_ARTICLE';
export const RESOLVING_ARTICLE = 'RESOLVING_ARTICLE';
export const GET_ARTICLE = 'GET_ARTICLE';
export const CLEAR_ARTICLE = 'CLEAR_ARTICLE';
export const INIT_RELOAD_ARTICLE = 'INIT_RELOAD_ARTICLE';
export const RELOAD_ARTICLE = 'RELOAD_ARTICLE';
export const INIT_CLEAR_DATA = 'INIT_CLEAR_DATA';
export const CLEAR_DATA = 'CLEAR_DATA';
export const INIT_SAVE_DATA = 'INIT_SAVE_DATA';
export const SAVE_DATA = 'SAVE_DATA';

export const INVALID_URL = 'INVALID_URL';
export const CLOSE_MODAL = 'CLOSE_MODAL';

export const ERROR_OCCURRED = 'ERROR_OCCURRED';
