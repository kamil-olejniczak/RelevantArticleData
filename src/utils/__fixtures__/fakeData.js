export const firstRecentlyVisitedSite = {
    id: 1,
    title: 'SomeTitle',
    url: 'SomeUrl',
    date: new Date(1 << 30),
};
export const secondRecentlyVisitedSite = {
    id: 2,
    title: 'SomeTitle2',
    url: 'SomeUrl2',
    date: new Date(1 << 30),
};
export const fakeRecentlyVisited = {
    recentlyVisited: [firstRecentlyVisitedSite, secondRecentlyVisitedSite]
};
export const fakeArticleData = {
    data: {
        title: 'Some Title',
        date_published: '1970-01-01',
        domain: 'www.dabliu.com',
        dek: 'Short dek text',
        excerpt: 'Short excerpt text',
        word_count: 111,
        lead_image_url: 'http://www.dabliu.pl/image.html',
    }
};
export const fakeTagStatistics = {
    div: 8,
    p: 2,
    a: 3,
    span: 10,
    i: 9,
    html: 1,
    br: 2,
    table: 7,
    hr: 6,
    input: 4,
    textarea: 5
};
export const fakeSortedTagStatistics = [
    'span', '10',
    'i', '9',
    'div', '8',
    'table', '7',
    'hr', '6',
    'textarea', '5',
    'input', '4',
    'a', '3',
    'br', '2',
    'p', '2',
];