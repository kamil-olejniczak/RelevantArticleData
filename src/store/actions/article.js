import * as actionTypes from './actionTypes';

export const articleResponse = ({data, content, recentlyVisited, statistics}, type) => ({
    type,
    data,
    content,
    recentlyVisited: recentlyVisited,
    statistics
});

export const getSpecifiedArticle = (url) => ({
    type: actionTypes.INIT_GET_ARTICLE,
    url
});

export const resolvingArticle = () => ({
    type: actionTypes.RESOLVING_ARTICLE
});

export const clearArticle = () => ({
    type: actionTypes.CLEAR_ARTICLE,
});

export const reloadPreviouslyVisitedArticle = (url) => ({
    type: actionTypes.INIT_RELOAD_ARTICLE,
    url
});

export const invalidUrl = () => ({
    type: actionTypes.INVALID_URL,
});

export const closeModalWithData = () => ({
    type: actionTypes.CLOSE_MODAL,
});

export const errorOccurred = (error) => ({
    type: actionTypes.ERROR_OCCURRED,
    error
});
