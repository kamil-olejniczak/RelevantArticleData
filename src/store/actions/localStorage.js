import * as actionTypes from './actionTypes';

export const clearData = () => ({
    type: actionTypes.INIT_CLEAR_DATA,
});

export const localStorageWasCleaned = () => ({
    type: actionTypes.CLEAR_DATA,
});

export const saveData = formattedData => ({
    type: actionTypes.INIT_SAVE_DATA,
    formattedData
});

export const localStorageWasUpdated = () => ({
    type: actionTypes.SAVE_DATA,
});