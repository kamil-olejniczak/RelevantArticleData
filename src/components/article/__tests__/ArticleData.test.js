import * as React from "react";
import ArticleData from '../ArticleData';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import * as actionTypes from '../../../store/actions/actionTypes';
import {fakeArticleData} from '../../../utils/__fixtures__/fakeData';

describe('ArticleData tests', () => {
    const mockStore = configureStore();

    let wrapper, store;

    beforeEach(() => {
        store = mockStore(fakeData);
        wrapper = shallow(<ArticleData store={store} data={fakeData.data}/>).dive();
    });

    it('renders correctly', () => {
        const component = renderer.create(
            <ArticleData store={store} data={fakeData.data}/>)
            .toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should not render div with article data when modal is closed', () => {
        wrapper = shallow(
            <ArticleData store={store} data={fakeData.data} isModalClosedByUser={true}/>).dive();

        expect(wrapper.find('div.ArticleData')).toHaveLength(0);
    });

    it('should render all paragraphs for provided data', () => {
        expect(wrapper.find('p')).toHaveLength(7);
    });

    it('should dispatch CLOSE_MODAL action when user dismisses modal', () => {
        wrapper = shallow(<ArticleData store={store} data={fakeData.data}/>).dive();
        wrapper.find('#close').simulate('click');

        const actions = store.getActions();
        const expectedPayload = { type: actionTypes.CLOSE_MODAL };

        expect(actions).toEqual([expectedPayload]);
    });
});

const fakeData = {
    ...fakeArticleData
};