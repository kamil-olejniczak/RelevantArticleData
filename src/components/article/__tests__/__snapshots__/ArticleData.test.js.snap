// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleData tests renders correctly 1`] = `
<div
  className="ArticleData"
>
  <span
    id="close"
    onClick={[Function]}
  >
    <i
      className="far fa-times-circle"
    />
  </span>
  <p>
    You are currently reading: "
    Some Title
    ".
  </p>
  <p>
    Which was published on 
    1970-01-01
    .
  </p>
  <p>
    On "
    www.dabliu.com
    " domain.
  </p>
  <p>
    Headline of the story (dek): "
    Short dek text
    ".
  </p>
  <p>
    Hand-crafted summary of content (excerpt): "
    Short excerpt text
    ".
  </p>
  <p>
    This article has: 
    111
     words.
  </p>
  <div>
    <p>
      Lead image to help you memorise it:
    </p>
    <img
      alt="Short excerpt text"
      className="leadImg"
      src="http://www.dabliu.pl/image.html"
    />
  </div>
</div>
`;
